t/security.t

```diff
@@ -0,0 +1,61 @@
+use Mojo::Base -strict;
+use Test::Mojo;
+use Test::More;
+
+use Mojolicious::Lite;
+post '/protected' => sub {
+  my $c = shift->openapi->valid_input or return;
+  $c->reply->openapi(200 => {ok => 1});
+  },
+  'protected';
+
+plugin OpenAPI => {
+  url      => 'data://main/sec.json',
+  security => {
+    dummy => sub {
+      my ($c, $config, $next) = @_;
+      return $c->$next if $c->req->headers->authorization;
+      return $c->render(json => $config, status => 401);
```

Comment:
"Using clear error descriptions please"..? What do you mean? You decide the document to render. You can do this if you like:

```
return $c->render(json => {sorry => "you don't have access"}, status => 401);
```

The way to mark the user as authenticated or authorised is by calling `$c->$next`